### PR TITLE
[Example] Add example of Canva Editor

### PR DESCRIPTION
**Pull request by davidalmarinho**
# Description

Add a new example, a canvas editor.
In a project that I am working on it, I spent some hours replicating the canvas editor of imgui example demo window in java because have to read cpp code since have to use List<ImVec2> intestead of ImVector<>. So, I think that is a example that is worth it having in this repo.

Fixes # (issue)
No fixes.

## Type of change

Please check options that are relevant.

- [x] Minor changes or tweaks (quality of life stuff)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
Didn't add tests, but tested the code and it is working properly.
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: example/src/main/java/ExampleCanvaEditor.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            //      ImGui.popStyleVar();
+            //      [...]
+            //      ImGui.endChild();
+
+            // Using InvisibleButton() as a convenience 1) it will advance the layout cursor and 2) allows us to use IsItemHovered()/IsItemActive()
+            ImVec2 canvasP0 = ImGui.getCursorScreenPos();      // ImDrawList API uses screen coordinates!
+            ImVec2 canvasSize = ImGui.getContentRegionAvail();   // Resize canvas to what's available
+
+            if (canvasSize.x < 50.0f) canvasSize.x = 50.0f;
+            if (canvasSize.y < 50.0f) canvasSize.y = 50.0f;
```
Comment (SpaiR) [inline example/src/main/java/ExampleCanvaEditor.java:65]: Can you fix a codestyle and remove oneline if's?

File: example/src/main/java/ExampleCanvaEditor.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            // Add first and second point
+            if (isHovered && !addingLine && ImGui.isMouseClicked(ImGuiMouseButton.Left)) {
+                pointList.add(mousePosInCanvas);
+                pointList.add(mousePosInCanvas);
+                addingLine = true;
+            }
+            if (addingLine) {
+                pointList.set(pointList.size() - 1, mousePosInCanvas);
+                if (!ImGui.isMouseDown(ImGuiMouseButton.Left))
+                    addingLine = false;
```
Comment (SpaiR) [inline example/src/main/java/ExampleCanvaEditor.java:96]: Also, always add brackets.

File: example/src/main/java/Extra.java [hunk 3]
```diff
@@ -26,6 +27,7 @@ public static void show(final Application app) {
         ImGui.checkbox("Show ImGuiColorTextEdit Demo Window", SHOW_IMGUI_COLOR_TEXT_EDIT_WINDOW);
         ImGui.checkbox("Show ImGuiFileDialog Demo Window", SHOW_IMGUI_FILE_DIALOG_WINDOW);
         ImGui.checkbox("Show ImGui MemoryEditor Demo Window", SHOW_IMGUI_MEMORY_EDITOR_WINDOW);
+        ImGui.checkbox("Show ImGui Canva Demo Window", SHOW_IMGUI_CANVA_EDITOR_WINDOW);
```
Comment (SpaiR) [inline example/src/main/java/Extra.java:30]: I believe we need to make the main window a little bit larger to remove a scrollbar.

File: example/src/main/java/Main.java [hunk 4]
```diff
@@ -70,6 +70,7 @@ private void initFonts(final ImGuiIO io) {
 
     @Override
     public void process() {
+        ImGui.setWindowSize(400, 410);
```
Comment (davidalmarinho) [inline example/src/main/java/Main.java:73]: Didn't find where the Demo's window height is set 🤔
But I think that we can change the window's size by doing this

**Comment by davidalmarinho** [comment]
You are welcome!
I think that now is all corrected.
If you need something else let me know :)

File: example/src/main/java/Main.java (hunk 4 above)
Comment (SpaiR) [reply example/src/main/java/Main.java:73]: You'll change a size of the next window, not the window with "extra" checkboxes. Wrap everything to their own window and change it.

**Review by SpaiR — CHANGES_REQUESTED** [review]

File: example/src/main/java/Main.java (hunk 4 above)
Comment (davidalmarinho) [reply example/src/main/java/Main.java]: Done.
Also added brackets to an if statement that I forgot.

**Review by SpaiR — APPROVED** [review]

**Pull request merged**